# This file is part of HiGFXback

REQUIRES="autotools-wrappers-ac-build expat-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69)
cat m4/* > aclocal.m4; autoconf; autoheader
./configure --enable-shared --with-system-expat --prefix=/

# build
make

# install
install -d $DESTDIR/bin
install python $DESTDIR/bin/python2
install -d $DESTDIR/include/python2
install -m 644 pyconfig.h $DESTDIR/include/python2
install -m 644 Include/abstract.h $DESTDIR/include/python2
install -m 644 Include/boolobject.h $DESTDIR/include/python2
install -m 644 Include/bufferobject.h $DESTDIR/include/python2
install -m 644 Include/bytearrayobject.h $DESTDIR/include/python2
install -m 644 Include/bytesobject.h $DESTDIR/include/python2
install -m 644 Include/cellobject.h $DESTDIR/include/python2
install -m 644 Include/ceval.h $DESTDIR/include/python2
install -m 644 Include/classobject.h $DESTDIR/include/python2
install -m 644 Include/cobject.h $DESTDIR/include/python2
install -m 644 Include/code.h $DESTDIR/include/python2
install -m 644 Include/codecs.h $DESTDIR/include/python2
install -m 644 Include/compile.h $DESTDIR/include/python2
install -m 644 Include/complexobject.h $DESTDIR/include/python2
install -m 644 Include/descrobject.h $DESTDIR/include/python2
install -m 644 Include/dictobject.h $DESTDIR/include/python2
install -m 644 Include/dtoa.h $DESTDIR/include/python2
install -m 644 Include/enumobject.h $DESTDIR/include/python2
install -m 644 Include/eval.h $DESTDIR/include/python2
install -m 644 Include/fileobject.h $DESTDIR/include/python2
install -m 644 Include/floatobject.h $DESTDIR/include/python2
install -m 644 Include/funcobject.h $DESTDIR/include/python2
install -m 644 Include/genobject.h $DESTDIR/include/python2
install -m 644 Include/import.h $DESTDIR/include/python2
install -m 644 Include/intobject.h $DESTDIR/include/python2
install -m 644 Include/intrcheck.h $DESTDIR/include/python2
install -m 644 Include/iterobject.h $DESTDIR/include/python2
install -m 644 Include/listobject.h $DESTDIR/include/python2
install -m 644 Include/longobject.h $DESTDIR/include/python2
install -m 644 Include/memoryobject.h $DESTDIR/include/python2
install -m 644 Include/methodobject.h $DESTDIR/include/python2
install -m 644 Include/modsupport.h $DESTDIR/include/python2
install -m 644 Include/moduleobject.h $DESTDIR/include/python2
install -m 644 Include/object.h $DESTDIR/include/python2
install -m 644 Include/objimpl.h $DESTDIR/include/python2
install -m 644 Include/patchlevel.h $DESTDIR/include/python2
install -m 644 Include/pyarena.h $DESTDIR/include/python2
install -m 644 Include/pycapsule.h $DESTDIR/include/python2
install -m 644 Include/pyctype.h $DESTDIR/include/python2
install -m 644 Include/pydebug.h $DESTDIR/include/python2
install -m 644 Include/pyerrors.h $DESTDIR/include/python2
install -m 644 Include/pyfpe.h $DESTDIR/include/python2
install -m 644 Include/pymacconfig.h $DESTDIR/include/python2
install -m 644 Include/pymath.h $DESTDIR/include/python2
install -m 644 Include/pymem.h $DESTDIR/include/python2
install -m 644 Include/pyport.h $DESTDIR/include/python2
install -m 644 Include/pystate.h $DESTDIR/include/python2
install -m 644 Include/pystrcmp.h $DESTDIR/include/python2
install -m 644 Include/pystrtod.h $DESTDIR/include/python2
install -m 644 Include/Python.h $DESTDIR/include/python2
install -m 644 Include/pythonrun.h $DESTDIR/include/python2
install -m 644 Include/rangeobject.h $DESTDIR/include/python2
install -m 644 Include/setobject.h $DESTDIR/include/python2
install -m 644 Include/sliceobject.h $DESTDIR/include/python2
install -m 644 Include/stringobject.h $DESTDIR/include/python2
install -m 644 Include/sysmodule.h $DESTDIR/include/python2
install -m 644 Include/traceback.h $DESTDIR/include/python2
install -m 644 Include/tupleobject.h $DESTDIR/include/python2
install -m 644 Include/unicodeobject.h $DESTDIR/include/python2
install -m 644 Include/warnings.h $DESTDIR/include/python2
install -m 644 Include/weakrefobject.h $DESTDIR/include/python2
ln -sf python2 $DESTDIR/include/python2.7
install -d $DESTDIR/lib
install libpython2.7.so.1 $DESTDIR/lib
ln -sf libpython2.7.so.1 $DESTDIR/lib/libpython2.7.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 Misc/python.pc $DESTDIR/lib/pkgconfig/python2.pc
install -d $DESTDIR/lib/python2
install -m 644 `cat pybuilddir.txt`/_sysconfigdata.py $DESTDIR/lib/python2
install -m 644 Lib/__future__.py $DESTDIR/lib/python2
install -m 644 Lib/_weakrefset.py $DESTDIR/lib/python2
install -m 644 Lib/_abcoll.py $DESTDIR/lib/python2
install -m 644 Lib/abc.py $DESTDIR/lib/python2
install -m 644 Lib/argparse.py $DESTDIR/lib/python2
install -m 644 Lib/base64.py $DESTDIR/lib/python2
install -m 644 Lib/codecs.py $DESTDIR/lib/python2
install -m 644 Lib/collections.py $DESTDIR/lib/python2
install -m 644 Lib/copy.py $DESTDIR/lib/python2
install -m 644 Lib/copy_reg.py $DESTDIR/lib/python2
install -m 644 Lib/fnmatch.py $DESTDIR/lib/python2
install -m 644 Lib/functools.py $DESTDIR/lib/python2
install -m 644 Lib/genericpath.py $DESTDIR/lib/python2
install -m 644 Lib/getopt.py $DESTDIR/lib/python2
install -m 644 Lib/gettext.py $DESTDIR/lib/python2
install -m 644 Lib/glob.py $DESTDIR/lib/python2
install -m 644 Lib/heapq.py $DESTDIR/lib/python2
install -m 644 Lib/io.py $DESTDIR/lib/python2
install -m 644 Lib/keyword.py $DESTDIR/lib/python2
install -m 644 Lib/linecache.py $DESTDIR/lib/python2
install -m 644 Lib/locale.py $DESTDIR/lib/python2
install -m 644 Lib/optparse.py $DESTDIR/lib/python2
install -m 644 Lib/os.py $DESTDIR/lib/python2
install -m 644 Lib/pickle.py $DESTDIR/lib/python2
install -m 644 Lib/posixpath.py $DESTDIR/lib/python2
install -m 644 Lib/py_compile.py $DESTDIR/lib/python2
install -m 644 Lib/re.py $DESTDIR/lib/python2
install -m 644 Lib/site.py $DESTDIR/lib/python2
install -m 644 Lib/socket.py $DESTDIR/lib/python2
install -m 644 Lib/sre_compile.py $DESTDIR/lib/python2
install -m 644 Lib/sre_constants.py $DESTDIR/lib/python2
install -m 644 Lib/sre_parse.py $DESTDIR/lib/python2
install -m 644 Lib/stat.py $DESTDIR/lib/python2
install -m 644 Lib/string.py $DESTDIR/lib/python2
install -m 644 Lib/StringIO.py $DESTDIR/lib/python2
install -m 644 Lib/struct.py $DESTDIR/lib/python2
install -m 644 Lib/subprocess.py $DESTDIR/lib/python2
install -m 644 Lib/sysconfig.py $DESTDIR/lib/python2
install -m 644 Lib/textwrap.py $DESTDIR/lib/python2
install -m 644 Lib/traceback.py $DESTDIR/lib/python2
install -m 644 Lib/types.py $DESTDIR/lib/python2
install -m 644 Lib/urllib.py $DESTDIR/lib/python2
install -m 644 Lib/urlparse.py $DESTDIR/lib/python2
install -m 644 Lib/UserDict.py $DESTDIR/lib/python2
install -m 644 Lib/warnings.py $DESTDIR/lib/python2
install -m 644 Lib/weakref.py $DESTDIR/lib/python2
install -d $DESTDIR/lib/python2/encodings
install -m 644 Lib/encodings/__init__.py $DESTDIR/lib/python2/encodings
install -m 644 Lib/encodings/aliases.py $DESTDIR/lib/python2/encodings
install -m 644 Lib/encodings/ascii.py $DESTDIR/lib/python2/encodings
install -d $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_collections.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_elementtree.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_functools.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_io.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_socket.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_struct.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/binascii.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/fcntl.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/itertools.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/math.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/operator.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/pyexpat.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/select.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/time.so $DESTDIR/lib/python2/lib-dynload
install -d $DESTDIR/lib/python2/xml
install -m 644 Lib/xml/__init__.py $DESTDIR/lib/python2/xml
install -d $DESTDIR/lib/python2/xml/etree
install -m 644 Lib/xml/etree/__init__.py $DESTDIR/lib/python2/xml/etree
install -m 644 Lib/xml/etree/cElementTree.py $DESTDIR/lib/python2/xml/etree
install -m 644 Lib/xml/etree/ElementTree.py $DESTDIR/lib/python2/xml/etree
install -d $DESTDIR/lib/python2/xml/parsers
install -m 644 Lib/xml/parsers/__init__.py $DESTDIR/lib/python2/xml/parsers
install -m 644 Lib/xml/parsers/expat.py $DESTDIR/lib/python2/xml/parsers
install -d $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/__init__.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/_exceptions.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/expatreader.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/handler.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/saxutils.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/xmlreader.py $DESTDIR/lib/python2/xml/sax
test -L $DESTDIR/lib/python2.7 || ln -s python2 $DESTDIR/lib/python2.7

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/python2-build.pc << EOF
Name: Python
Version: 2.7.18
Description: Python programming language
Requires: $REQUIRES

devel=\\
/include/python2/abstract.h \\
/include/python2/boolobject.h \\
/include/python2/bufferobject.h \\
/include/python2/bytearrayobject.h \\
/include/python2/bytesobject.h \\
/include/python2/cellobject.h \\
/include/python2/ceval.h \\
/include/python2/classobject.h \\
/include/python2/cobject.h \\
/include/python2/code.h \\
/include/python2/codecs.h \\
/include/python2/compile.h \\
/include/python2/complexobject.h \\
/include/python2/descrobject.h \\
/include/python2/dictobject.h \\
/include/python2/dtoa.h \\
/include/python2/enumobject.h \\
/include/python2/eval.h \\
/include/python2/fileobject.h \\
/include/python2/floatobject.h \\
/include/python2/funcobject.h \\
/include/python2/genobject.h \\
/include/python2/import.h \\
/include/python2/intobject.h \\
/include/python2/intrcheck.h \\
/include/python2/iterobject.h \\
/include/python2/listobject.h \\
/include/python2/longobject.h \\
/include/python2/memoryobject.h \\
/include/python2/methodobject.h \\
/include/python2/modsupport.h \\
/include/python2/moduleobject.h \\
/include/python2/object.h \\
/include/python2/objimpl.h \\
/include/python2/patchlevel.h \\
/include/python2/pyarena.h \\
/include/python2/pycapsule.h \\
/include/python2/pyconfig.h \\
/include/python2/pyctype.h \\
/include/python2/pydebug.h \\
/include/python2/pyerrors.h \\
/include/python2/pyfpe.h \\
/include/python2/pymacconfig.h \\
/include/python2/pymath.h \\
/include/python2/pymem.h \\
/include/python2/pyport.h \\
/include/python2/pystate.h \\
/include/python2/pystrcmp.h \\
/include/python2/pystrtod.h \\
/include/python2/Python.h \\
/include/python2/pythonrun.h \\
/include/python2/rangeobject.h \\
/include/python2/setobject.h \\
/include/python2/sliceobject.h \\
/include/python2/stringobject.h \\
/include/python2/sysmodule.h \\
/include/python2/traceback.h \\
/include/python2/tupleobject.h \\
/include/python2/unicodeobject.h \\
/include/python2/warnings.h \\
/include/python2/weakrefobject.h \\
/include/python2.7 \\
/lib/libpython2.7.so \\
/lib/pkgconfig/python2.pc

exec=\\
/bin/python2 \\
/lib/libpython2.7.so.1 \\
/lib/python2/__future__.py \\
/lib/python2/_abcoll.py \\
/lib/python2/_sysconfigdata.py \\
/lib/python2/_weakrefset.py \\
/lib/python2/abc.py \\
/lib/python2/argparse.py \\
/lib/python2/base64.py \\
/lib/python2/codecs.py \\
/lib/python2/collections.py \\
/lib/python2/copy.py \\
/lib/python2/copy_reg.py \\
/lib/python2/fnmatch.py \\
/lib/python2/functools.py \\
/lib/python2/genericpath.py \\
/lib/python2/getopt.py \\
/lib/python2/gettext.py \\
/lib/python2/glob.py \\
/lib/python2/heapq.py \\
/lib/python2/io.py.py \\
/lib/python2/keyword.py \\
/lib/python2/linecache.py \\
/lib/python2/locale.py \\
/lib/python2/optparse.py \\
/lib/python2/os.py \\
/lib/python2/pickle.py \\
/lib/python2/posixpath.py \\
/lib/python2/py_compile.py \\
/lib/python2/re.py \\
/lib/python2/site.py \\
/lib/python2/socket.py \\
/lib/python2/sre_compile.py \\
/lib/python2/sre_constants.py \\
/lib/python2/sre_parse.py \\
/lib/python2/stat.py \\
/lib/python2/string.py \\
/lib/python2/StringIO.py \\
/lib/python2/struct.py \\
/lib/python2/subprocess.py \\
/lib/python2/sysconfig.py \\
/lib/python2/textwrap.py \\
/lib/python2/traceback.py \\
/lib/python2/types.py \\
/lib/python2/urllib.py \\
/lib/python2/urlparse.py \\
/lib/python2/UserDict.py \\
/lib/python2/warnings.py \\
/lib/python2/weakref.py \\
/lib/python2/encodings/__init__.py \\
/lib/python2/encodings/aliases.py \\
/lib/python2/encodings/ascii.py \\
/lib/python2/lib-dynload/_collections.so \\
/lib/python2/lib-dynload/_elementtree.so \\
/lib/python2/lib-dynload/_functools.so \\
/lib/python2/lib-dynload/_io.so \\
/lib/python2/lib-dynload/_socket.so \\
/lib/python2/lib-dynload/_struct.so \\
/lib/python2/lib-dynload/binascii.so \\
/lib/python2/lib-dynload/fcntl.so \\
/lib/python2/lib-dynload/itertools.so \\
/lib/python2/lib-dynload/math.so \\
/lib/python2/lib-dynload/operator.so \\
/lib/python2/lib-dynload/pyexpat.so \\
/lib/python2/lib-dynload/select.so \\
/lib/python2/lib-dynload/time.so \\
/lib/python2/xml/__init__.py \\
/lib/python2/xml/etree/__init__.py \\
/lib/python2/xml/etree/cElementTree.py \\
/lib/python2/xml/etree/ElementTree.py \\
/lib/python2/xml/parsers/__init__.py \\
/lib/python2/xml/parsers/expat.py \\
/lib/python2/xml/sax/__init__.py \\
/lib/python2/xml/sax/_exceptions.py \\
/lib/python2/xml/sax/expatreader.py \\
/lib/python2/xml/sax/handler.py \\
/lib/python2/xml/sax/saxutils.py \\
/lib/python2/xml/sax/xmlreader.py \\
/lib/python2.7
EOF

exit
--- Python-2.7.18.orig/configure.ac
+++ Python-2.7.18/configure.ac
@@ -79,7 +79,7 @@
 VERSION=PYTHON_VERSION
 
 AC_SUBST(SOVERSION)
-SOVERSION=1.0
+SOVERSION=1
 
 # The later defininition of _XOPEN_SOURCE disables certain features
 # on Linux, so we need _GNU_SOURCE to re-enable them (makedev, tm_zone).
--- Python-2.7.18.orig/m4/pkg.m4
+++ Python-2.7.18/m4/pkg.m4
@@ -0,0 +1,275 @@
+dnl pkg.m4 - Macros to locate and utilise pkg-config.   -*- Autoconf -*-
+dnl serial 11 (pkg-config-0.29.1)
+dnl
+dnl Copyright © 2004 Scott James Remnant <scott@netsplit.com>.
+dnl Copyright © 2012-2015 Dan Nicholson <dbn.lists@gmail.com>
+dnl
+dnl This program is free software; you can redistribute it and/or modify
+dnl it under the terms of the GNU General Public License as published by
+dnl the Free Software Foundation; either version 2 of the License, or
+dnl (at your option) any later version.
+dnl
+dnl This program is distributed in the hope that it will be useful, but
+dnl WITHOUT ANY WARRANTY; without even the implied warranty of
+dnl MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+dnl General Public License for more details.
+dnl
+dnl You should have received a copy of the GNU General Public License
+dnl along with this program; if not, write to the Free Software
+dnl Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+dnl 02111-1307, USA.
+dnl
+dnl As a special exception to the GNU General Public License, if you
+dnl distribute this file as part of a program that contains a
+dnl configuration script generated by Autoconf, you may include it under
+dnl the same distribution terms that you use for the rest of that
+dnl program.
+
+dnl PKG_PREREQ(MIN-VERSION)
+dnl -----------------------
+dnl Since: 0.29
+dnl
+dnl Verify that the version of the pkg-config macros are at least
+dnl MIN-VERSION. Unlike PKG_PROG_PKG_CONFIG, which checks the user's
+dnl installed version of pkg-config, this checks the developer's version
+dnl of pkg.m4 when generating configure.
+dnl
+dnl To ensure that this macro is defined, also add:
+dnl m4_ifndef([PKG_PREREQ],
+dnl     [m4_fatal([must install pkg-config 0.29 or later before running autoconf/autogen])])
+dnl
+dnl See the "Since" comment for each macro you use to see what version
+dnl of the macros you require.
+m4_defun([PKG_PREREQ],
+[m4_define([PKG_MACROS_VERSION], [0.29.1])
+m4_if(m4_version_compare(PKG_MACROS_VERSION, [$1]), -1,
+    [m4_fatal([pkg.m4 version $1 or higher is required but ]PKG_MACROS_VERSION[ found])])
+])dnl PKG_PREREQ
+
+dnl PKG_PROG_PKG_CONFIG([MIN-VERSION])
+dnl ----------------------------------
+dnl Since: 0.16
+dnl
+dnl Search for the pkg-config tool and set the PKG_CONFIG variable to
+dnl first found in the path. Checks that the version of pkg-config found
+dnl is at least MIN-VERSION. If MIN-VERSION is not specified, 0.9.0 is
+dnl used since that's the first version where most current features of
+dnl pkg-config existed.
+AC_DEFUN([PKG_PROG_PKG_CONFIG],
+[m4_pattern_forbid([^_?PKG_[A-Z_]+$])
+m4_pattern_allow([^PKG_CONFIG(_(PATH|LIBDIR|SYSROOT_DIR|ALLOW_SYSTEM_(CFLAGS|LIBS)))?$])
+m4_pattern_allow([^PKG_CONFIG_(DISABLE_UNINSTALLED|TOP_BUILD_DIR|DEBUG_SPEW)$])
+AC_ARG_VAR([PKG_CONFIG], [path to pkg-config utility])
+AC_ARG_VAR([PKG_CONFIG_PATH], [directories to add to pkg-config's search path])
+AC_ARG_VAR([PKG_CONFIG_LIBDIR], [path overriding pkg-config's built-in search path])
+
+if test "x$ac_cv_env_PKG_CONFIG_set" != "xset"; then
+	AC_PATH_TOOL([PKG_CONFIG], [pkg-config])
+fi
+if test -n "$PKG_CONFIG"; then
+	_pkg_min_version=m4_default([$1], [0.9.0])
+	AC_MSG_CHECKING([pkg-config is at least version $_pkg_min_version])
+	if $PKG_CONFIG --atleast-pkgconfig-version $_pkg_min_version; then
+		AC_MSG_RESULT([yes])
+	else
+		AC_MSG_RESULT([no])
+		PKG_CONFIG=""
+	fi
+fi[]dnl
+])dnl PKG_PROG_PKG_CONFIG
+
+dnl PKG_CHECK_EXISTS(MODULES, [ACTION-IF-FOUND], [ACTION-IF-NOT-FOUND])
+dnl -------------------------------------------------------------------
+dnl Since: 0.18
+dnl
+dnl Check to see whether a particular set of modules exists. Similar to
+dnl PKG_CHECK_MODULES(), but does not set variables or print errors.
+dnl
+dnl Please remember that m4 expands AC_REQUIRE([PKG_PROG_PKG_CONFIG])
+dnl only at the first occurence in configure.ac, so if the first place
+dnl it's called might be skipped (such as if it is within an "if", you
+dnl have to call PKG_CHECK_EXISTS manually
+AC_DEFUN([PKG_CHECK_EXISTS],
+[AC_REQUIRE([PKG_PROG_PKG_CONFIG])dnl
+if test -n "$PKG_CONFIG" && \
+    AC_RUN_LOG([$PKG_CONFIG --exists --print-errors "$1"]); then
+  m4_default([$2], [:])
+m4_ifvaln([$3], [else
+  $3])dnl
+fi])
+
+dnl _PKG_CONFIG([VARIABLE], [COMMAND], [MODULES])
+dnl ---------------------------------------------
+dnl Internal wrapper calling pkg-config via PKG_CONFIG and setting
+dnl pkg_failed based on the result.
+m4_define([_PKG_CONFIG],
+[if test -n "$$1"; then
+    pkg_cv_[]$1="$$1"
+ elif test -n "$PKG_CONFIG"; then
+    PKG_CHECK_EXISTS([$3],
+                     [pkg_cv_[]$1=`$PKG_CONFIG --[]$2 "$3" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes ],
+		     [pkg_failed=yes])
+ else
+    pkg_failed=untried
+fi[]dnl
+])dnl _PKG_CONFIG
+
+dnl _PKG_SHORT_ERRORS_SUPPORTED
+dnl ---------------------------
+dnl Internal check to see if pkg-config supports short errors.
+AC_DEFUN([_PKG_SHORT_ERRORS_SUPPORTED],
+[AC_REQUIRE([PKG_PROG_PKG_CONFIG])
+if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
+        _pkg_short_errors_supported=yes
+else
+        _pkg_short_errors_supported=no
+fi[]dnl
+])dnl _PKG_SHORT_ERRORS_SUPPORTED
+
+
+dnl PKG_CHECK_MODULES(VARIABLE-PREFIX, MODULES, [ACTION-IF-FOUND],
+dnl   [ACTION-IF-NOT-FOUND])
+dnl --------------------------------------------------------------
+dnl Since: 0.4.0
+dnl
+dnl Note that if there is a possibility the first call to
+dnl PKG_CHECK_MODULES might not happen, you should be sure to include an
+dnl explicit call to PKG_PROG_PKG_CONFIG in your configure.ac
+AC_DEFUN([PKG_CHECK_MODULES],
+[AC_REQUIRE([PKG_PROG_PKG_CONFIG])dnl
+AC_ARG_VAR([$1][_CFLAGS], [C compiler flags for $1, overriding pkg-config])dnl
+AC_ARG_VAR([$1][_LIBS], [linker flags for $1, overriding pkg-config])dnl
+
+pkg_failed=no
+AC_MSG_CHECKING([for $1])
+
+_PKG_CONFIG([$1][_CFLAGS], [cflags], [$2])
+_PKG_CONFIG([$1][_LIBS], [libs], [$2])
+
+m4_define([_PKG_TEXT], [Alternatively, you may set the environment variables $1[]_CFLAGS
+and $1[]_LIBS to avoid the need to call pkg-config.
+See the pkg-config man page for more details.])
+
+if test $pkg_failed = yes; then
+   	AC_MSG_RESULT([no])
+        _PKG_SHORT_ERRORS_SUPPORTED
+        if test $_pkg_short_errors_supported = yes; then
+	        $1[]_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "$2" 2>&1`
+        else 
+	        $1[]_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "$2" 2>&1`
+        fi
+	# Put the nasty error message in config.log where it belongs
+	echo "$$1[]_PKG_ERRORS" >&AS_MESSAGE_LOG_FD
+
+	m4_default([$4], [AC_MSG_ERROR(
+[Package requirements ($2) were not met:
+
+$$1_PKG_ERRORS
+
+Consider adjusting the PKG_CONFIG_PATH environment variable if you
+installed software in a non-standard prefix.
+
+_PKG_TEXT])[]dnl
+        ])
+elif test $pkg_failed = untried; then
+     	AC_MSG_RESULT([no])
+	m4_default([$4], [AC_MSG_FAILURE(
+[The pkg-config script could not be found or is too old.  Make sure it
+is in your PATH or set the PKG_CONFIG environment variable to the full
+path to pkg-config.
+
+_PKG_TEXT
+
+To get pkg-config, see <http://pkg-config.freedesktop.org/>.])[]dnl
+        ])
+else
+	$1[]_CFLAGS=$pkg_cv_[]$1[]_CFLAGS
+	$1[]_LIBS=$pkg_cv_[]$1[]_LIBS
+        AC_MSG_RESULT([yes])
+	$3
+fi[]dnl
+])dnl PKG_CHECK_MODULES
+
+
+dnl PKG_CHECK_MODULES_STATIC(VARIABLE-PREFIX, MODULES, [ACTION-IF-FOUND],
+dnl   [ACTION-IF-NOT-FOUND])
+dnl ---------------------------------------------------------------------
+dnl Since: 0.29
+dnl
+dnl Checks for existence of MODULES and gathers its build flags with
+dnl static libraries enabled. Sets VARIABLE-PREFIX_CFLAGS from --cflags
+dnl and VARIABLE-PREFIX_LIBS from --libs.
+dnl
+dnl Note that if there is a possibility the first call to
+dnl PKG_CHECK_MODULES_STATIC might not happen, you should be sure to
+dnl include an explicit call to PKG_PROG_PKG_CONFIG in your
+dnl configure.ac.
+AC_DEFUN([PKG_CHECK_MODULES_STATIC],
+[AC_REQUIRE([PKG_PROG_PKG_CONFIG])dnl
+_save_PKG_CONFIG=$PKG_CONFIG
+PKG_CONFIG="$PKG_CONFIG --static"
+PKG_CHECK_MODULES($@)
+PKG_CONFIG=$_save_PKG_CONFIG[]dnl
+])dnl PKG_CHECK_MODULES_STATIC
+
+
+dnl PKG_INSTALLDIR([DIRECTORY])
+dnl -------------------------
+dnl Since: 0.27
+dnl
+dnl Substitutes the variable pkgconfigdir as the location where a module
+dnl should install pkg-config .pc files. By default the directory is
+dnl $libdir/pkgconfig, but the default can be changed by passing
+dnl DIRECTORY. The user can override through the --with-pkgconfigdir
+dnl parameter.
+AC_DEFUN([PKG_INSTALLDIR],
+[m4_pushdef([pkg_default], [m4_default([$1], ['${libdir}/pkgconfig'])])
+m4_pushdef([pkg_description],
+    [pkg-config installation directory @<:@]pkg_default[@:>@])
+AC_ARG_WITH([pkgconfigdir],
+    [AS_HELP_STRING([--with-pkgconfigdir], pkg_description)],,
+    [with_pkgconfigdir=]pkg_default)
+AC_SUBST([pkgconfigdir], [$with_pkgconfigdir])
+m4_popdef([pkg_default])
+m4_popdef([pkg_description])
+])dnl PKG_INSTALLDIR
+
+
+dnl PKG_NOARCH_INSTALLDIR([DIRECTORY])
+dnl --------------------------------
+dnl Since: 0.27
+dnl
+dnl Substitutes the variable noarch_pkgconfigdir as the location where a
+dnl module should install arch-independent pkg-config .pc files. By
+dnl default the directory is $datadir/pkgconfig, but the default can be
+dnl changed by passing DIRECTORY. The user can override through the
+dnl --with-noarch-pkgconfigdir parameter.
+AC_DEFUN([PKG_NOARCH_INSTALLDIR],
+[m4_pushdef([pkg_default], [m4_default([$1], ['${datadir}/pkgconfig'])])
+m4_pushdef([pkg_description],
+    [pkg-config arch-independent installation directory @<:@]pkg_default[@:>@])
+AC_ARG_WITH([noarch-pkgconfigdir],
+    [AS_HELP_STRING([--with-noarch-pkgconfigdir], pkg_description)],,
+    [with_noarch_pkgconfigdir=]pkg_default)
+AC_SUBST([noarch_pkgconfigdir], [$with_noarch_pkgconfigdir])
+m4_popdef([pkg_default])
+m4_popdef([pkg_description])
+])dnl PKG_NOARCH_INSTALLDIR
+
+
+dnl PKG_CHECK_VAR(VARIABLE, MODULE, CONFIG-VARIABLE,
+dnl [ACTION-IF-FOUND], [ACTION-IF-NOT-FOUND])
+dnl -------------------------------------------
+dnl Since: 0.28
+dnl
+dnl Retrieves the value of the pkg-config variable for the given module.
+AC_DEFUN([PKG_CHECK_VAR],
+[AC_REQUIRE([PKG_PROG_PKG_CONFIG])dnl
+AC_ARG_VAR([$1], [value of $3 for $2, overriding pkg-config])dnl
+
+_PKG_CONFIG([$1], [variable="][$3]["], [$2])
+AS_VAR_COPY([$1], [pkg_cv_][$1])
+
+AS_VAR_IF([$1], [""], [$5], [$4])dnl
+])dnl PKG_CHECK_VAR
